go/exoservice/exoservice.go

```diff
@@ -1 +1,77 @@
 package exoservice
+
+import (
+	"fmt"
+
+	"github.com/Originate/exocom/go/exorelay"
+	"github.com/Originate/exocom/go/structs"
+)
+
+// SendHelpers contains helper functions for sending messages
+type SendHelpers struct {
+	Reply func(exorelay.MessageOptions)
+}
+
+// MessageHandler is the function signature for handling a message
+type MessageHandler func(structs.MessagePayload, SendHelpers)
+
+// MessageHandlerMapping is a map from message name to MessageHandler
+type MessageHandlerMapping map[string]MessageHandler
+
+// ExoService is the high level Go API to talk to Exocom
+type ExoService struct {
+	Config          exorelay.Config
+	MessageHandlers MessageHandlerMapping
+	exoRelay        exorelay.ExoRelay
+}
+
+// Connect brings an ExoService instance online
+func (e *ExoService) Connect() error {
+	e.exoRelay = exorelay.ExoRelay{Config: e.Config}
+	err := e.exoRelay.Connect()
+	if err != nil {
+		return err
+	}
+	e.listenForMessages()
+	return nil
+}
+
+// Helpers
+
+func (e *ExoService) listenForMessages() {
+	messageChannel := e.exoRelay.GetMessageChannel()
+	go func() {
```

Comment:
I think this function should block. the `listenForMessages` function in exorelay blocks below and so does `http.ListenAndServe`. I think it's reasonable if you want the exposed `Connect` function to not block, but then we should break out the goroutine there.

```go
go e.listenForMessages()
```